src/frontend/src/components/SearchPage/SearchPage.jsx

```diff
@@ -62,38 +71,107 @@ const SearchPage = () => {
             },
           };
         });
-      setResults(finalResults);
+      setResults({ type: 'author', searchResults: finalResults });
     },
   });
 
   // Hook that will re-render the page only if the state of searchText changes
   useEffect(() => {
-    executeSearch();
-  }, [searchText]);
+    async function searchPosts() {
+      // Setting to check length > 3, as the component will trying to fetch with 0 length causing error
+      if (searchText.length > 3) {
+        const encodedSearchText = encodeURIComponent(searchText);
+        try {
+          setFetchLoading(true);
+          let res = await fetch(`${telescopeUrl}/query?search=${encodedSearchText}`);
+          if (!res.ok) {
+            throw new Error(res.statusText);
+          }
+          let contentType = res.headers.get('content-type');
+          if (!contentType || !contentType.includes('application/json')) {
+            throw new Error('Response was not json');
+          }
+          const searchResults = await res.json();
+          // ES values property contains an array of objects with a (feed) id property
+          const postIds = searchResults.values.map((result) => result.id);
+          const posts = await Promise.all(
+            postIds.map(async (id) => {
+              res = await fetch(`${telescopeUrl}/posts/${id}`);
+              if (!res.ok) {
+                throw new Error(res.statusText);
+              }
+              contentType = res.headers.get('content-type');
+              if (!contentType || !contentType.includes('application/json')) {
+                throw new Error('Response was not json');
+              }
+              return res.json();
+            })
+          );
+          setFetchLoading(false);
+          await setResults({ type: 'post', searchResults: posts });
+        } catch (error) {
+          console.log('Something went wrong while fetching data', error);
+        }
+      } else {
+        setResults({ type: 'post', searchResults: [] });
+      }
+    }
+
+    if (filter === 'author') {
+      executeSearch();
```

Comment:
Would `searchAuthors()` be a more logical name for this, given `searchPosts()` below?

Reply:
renamed